src/main/scala/nl.knaw.dans.easy.deposit/StateManager.scala

```diff
@@ -0,0 +1,141 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.deposit
+
+import java.net.URL
+import java.util.UUID
+
+import better.files.File
+import nl.knaw.dans.easy.deposit.Errors.{ IllegalStateTransitionException, InvalidPropertyException, PropertyNotFoundException }
+import nl.knaw.dans.easy.deposit.docs.StateInfo
+import nl.knaw.dans.easy.deposit.docs.StateInfo.State
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+import org.apache.commons.configuration.PropertiesConfiguration
+
+import scala.util.{ Failure, Success, Try }
+
+case class StateManager(depositDir: File, submitBase: File, landingPageBase: URL) extends DebugEnhancedLogging {
+
+  private val stateDescriptionKey = "state.description"
+  private val stateLabelKey = "state.label"
+
+  /** created/destroyed when the state changes, needed to peek into SUBMITTED/UUID/deposit.properties */
+  private val bagIdKey = "bag-store.bag-id"
+
+  /** Available as side effect for properties not related to fetching or updating the state of the deposit. */
+  val draftProps = new PropertiesConfiguration(
+    (depositDir / "deposit.properties").toJava
+  )
+  private lazy val submittedProps = new PropertiesConfiguration(
+    (submitBase / getProp(bagIdKey, draftProps) / "deposit.properties").toJava
+  )
+
+  /** @return the state-label/description from drafts/USER/UUID/deposit.properties
+   *          unless more recent values might be available in SUBMITTED/UUID/deposit.properties
+   */
+  def getStateInfo: Try[StateInfo] = Try {
+    toDraftState(getProp(stateLabelKey, draftProps)) match {
+      case State.submitted | State.inProgress =>
+        val newState: StateInfo = getProp(stateLabelKey, submittedProps) match {
+          case "SUBMITTED" => StateInfo(State.submitted, getStateDescription(draftProps))
+          case "REJECTED" => StateInfo(State.rejected, getStateDescription(submittedProps))
+          case "FAILED" => StateInfo(State.inProgress, s"The deposit is in progress.")
+          case "IN_REVIEW" => StateInfo(State.inProgress, s"The deposit is available at $landingPage")
+          case "FEDORA_ARCHIVED" | "ARCHIVED" => StateInfo(State.archived, s"The dataset is published at $landingPage")
+          case str: String =>
+            logger.error(InvalidPropertyException(stateLabelKey, str, submittedProps).getMessage)
+            StateInfo(State.inProgress, s"The deposit is in progress.")
+        }
+        saveNewState(newState)
+        newState
+      case draftState @ (State.draft | State.rejected | State.archived) =>
```

Comment:
Once the state became archived, the landing page persists. Even when the service is restarted with another base URL in `application.properties`